src/App.tsx

```diff
@@ -1,34 +1,111 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
+import { StatusFilter, TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  return (
-    <>
-      <div className="section">
-        <div className="container">
-          <div className="box">
-            <h1 className="title">Todos:</h1>
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [statusFilter, setStatusFilter] = useState<StatusFilter>('all');
+  const [query, setQuery] = useState('');
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
 
-            <div className="block">
-              <TodoFilter />
-            </div>
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+        setIsLoading(false);
+      })
+      .catch(() => setIsError(true));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    if (!todos) {
+      return null;
+    }
+
+    let newFilteredTodos;
+
+    const includesQuery = (todo: Todo) =>
+      query
+        ? todo.title.toLowerCase().includes(query.trim().toLowerCase())
+        : true;
+
+    switch (statusFilter) {
+      case 'active':
+        newFilteredTodos = todos.filter(t => !t.completed && includesQuery(t));
+        break;
+      case 'completed':
+        newFilteredTodos = todos.filter(t => t.completed && includesQuery(t));
+        break;
+      case 'all':
+      default:
+        newFilteredTodos = todos.filter(t => includesQuery(t));
+    }
```

Comment:
I think I would rather use `useMemo` and just return `newFileteredTodos` instead of reassigning them